dbms/src/Storages/Page/V3/BlobStore.cpp

```diff
@@ -561,10 +567,74 @@ void BlobStore::read(BlobFileId blob_id, BlobFileOffset offset, char * buffers,
 }
 
 
+struct BlobStoreGCInfo
+{
+    String toString() const
+    {
+        return fmt::format("{}. {}. {}. {}. ",
+                           toTypeString("Read-Only Blob", 0),
+                           toTypeString("No GC Blob", 1),
+                           toTypeString("Full GC Blob", 2),
+                           toTypeString("Truncated Blob", 3));
+    }
+
+    void appendToReadOnlyBlob(const BlobFileId blob_id, double valid_rate)
+    {
+        blob_gc_info[0].emplace_back(std::make_pair(blob_id, valid_rate));
+    }
+
+    void appendToNoNeedGCBlob(const BlobFileId blob_id, double valid_rate)
+    {
+        blob_gc_info[1].emplace_back(std::make_pair(blob_id, valid_rate));
+    }
+
+    void appendToNeedGCBlob(const BlobFileId blob_id, double valid_rate)
+    {
+        blob_gc_info[2].emplace_back(std::make_pair(blob_id, valid_rate));
+    }
+
+    void appendToTruncatedBlob(const BlobFileId blob_id, double valid_rate)
+    {
+        blob_gc_info[3].emplace_back(std::make_pair(blob_id, valid_rate));
+    }
+
+private:
+    // 1. read only blob
+    // 2. no need gc blob
+    // 3. full gc blob
+    // 4. need truncate blob
+    std::vector<std::pair<BlobFileId, double>> blob_gc_info[4];
+
+    String toTypeString(const String prefix, const size_t index) const
```

Comment:
Maybe this is better for performance since all the input param are `const char *` in actual
```suggestion
    String toTypeString(const std::string_view prefix, const size_t index) const
```